### PR TITLE
Update getCustomer method in order class

### PR DESCRIPTION
**Pull request by sertlab**
### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/25268 $order->getCustomer() returns NULL for registered customer

### Manual testing scenarios (*)
- Create an observer with `sales_order_save_after` event
- In execute method of the observer try to use getCustomer() method

**Expected Result**
A customer object should be returned

**Actual Result**
Null Returned

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by orlangur — CHANGES_REQUESTED** [review]
Such change is not backward compatible, see https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/

What problem are you trying to solve by such change?

**Comment by ihor-sviziev** [comment]
I think this is related issue https://github.com/magento/magento2/issues/25268
I think the most correct fix will be updating phpdoc block for getCustomer method to show that it could return null.
https://github.com/magento/magento2/issues/25268#issuecomment-559677023

**Comment by sertlab** [comment]
> I think this is related issue #25268
> I think the most correct fix will be updating phpdoc block for getCustomer method to show that it could return null.
> [#25268 (comment)](https://github.com/magento/magento2/issues/25268#issuecomment-559677023)

I tested and it returns only null.
By using this fix we get the correct customer object from order.

**Comment by ihor-sviziev** [comment]
@sertlab it might return customer object in some specific case.
Getting customer is not direct responsibility of data Order interface, it should contains only data about order, and ids of related entities.
Also both methods getCustomer and getCustomerId could return null in case if order was placed as guest, and that’s absolutely valid case.
I raised this discussion in the original issue that you wanted to fix. For now will put this PR on hold

**Comment by sertlab** [comment]
> @sertlab it might return customer object in some specific case.
> Getting customer is not direct responsibility of data Order interface, it should contains only data about order, and ids of related entities.
> Also both methods getCustomer and getCustomerId could return null in case if order was placed as guest, and that’s absolutely valid case.
> I raised this discussion in the original issue that you wanted to fix. For now will put this PR on hold

Hi . @ihor-sviziev  Tested with registered and logged in customer and getCustomer() still returns null in observer

**Comment by ihor-sviziev** [comment]
> Hi . @ihor-sviziev Tested with registered and logged in customer and getCustomer() still returns null in observer

Yeah, this is expected behavior! Data object should just be a container for data, it should not contains any business logic, for example retrieving customer model.

Fix actually should be just changing phpdoc block that method getCustomer could return null. In ideal situation would be great just remove it from that model at all, but I believe it will be treated as backward incompatible change, so let’s not do that.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @sertlab,
Could you rollback your changes in OrderInterface, Order classes and just update phpdoc block in Order model in order to highlight that it could return null?

**Comment by sertlab** [comment]
@ihor-sviziev  Hi. It's now updated as per your instructions

File: app/code/Magento/Sales/Model/Order.php [hunk 1]
```diff
@@ -1326,7 +1326,7 @@ public function getTrackingNumbers()
      */
     public function getShippingMethod($asObject = false)
     {
-        $shippingMethod = parent::getShippingMethod();
```
Comment (ihor-sviziev) [inline app/code/Magento/Sales/Model/Order.php:1329]: Clarification why it was changed: static test was complaining that parent method `getShippingMethod` doesn't exists. I checked - it's really doesn't exists, so instead it goes to magic method `__call` and runs `getData` method

Comment (sertlab) [reply app/code/Magento/Sales/Model/Order.php:1329]: Just fixed and all back to normal

Comment (ihor-sviziev) [reply app/code/Magento/Sales/Model/Order.php:1329]: @sertlab 
That was actually fix for static tests, that I've commited to your branch :)
After your commit "roll back changes" we'll have failing static tests again.

Could you revert your commit "roll back changes", squash all changes and force push?

Comment (sertlab) [reply app/code/Magento/Sales/Model/Order.php:1329]: Ok my bad. it's now reverted. i was thinking that i messed up the code during changes, without realising that you commit it more changes to make the tests pass.
Sry and thanks !

**Review by ihor-sviziev — APPROVED** [review]
Failing functional & unit tests are not related to changes from this PR, so I'm approving this PR

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6809](https://jira.corp.magento.com/browse/ENGCOM-6809) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**